vivisect/__init__.py

```diff
@@ -1129,8 +1129,19 @@ def parseOpcode(self, va, arch=envi.ARCH_DEFAULT):
             # so that at least parse opcode wont fail
             if loctup is not None and loctup[L_TINFO] and loctup[L_LTYPE] == LOC_OP:
                 arch = loctup[L_TINFO]
+        key = (va, arch, b[:16])
+        valu = self._op_cache.get(key, None)
+        if valu and not skip:
+            return valu
```

Comment:
```suggestion
        if not skip_cache:
            key = (va, arch, b[:16])
            valu = self._op_cache.get(key, None)
            if valu:
                return valu
```

Reply:
Good suggestion. Though I'm not a fan of the double check, so I pushed up a modified version